src/compiler/scala/tools/nsc/transform/Erasure.scala

```diff
@@ -1216,14 +1216,23 @@ abstract class Erasure extends InfoTransform
           Match(Typed(selector, TypeTree(selector.tpe)), cases)
 
         case Literal(ct) =>
-          // We remove the original tree attachments in pre-easure to free up memory
+          // We remove the original tree attachments in pre-erasure to free up memory
           val cleanLiteral = tree.removeAttachment[OriginalTreeAttachment]
 
           if (ct.tag == ClazzTag && ct.typeValue.typeSymbol != definitions.UnitClass) {
-            val erased = ct.typeValue.dealiasWiden match {
-              case tr @ TypeRef(_, clazz, _) if clazz.isDerivedValueClass => scalaErasure.eraseNormalClassRef(tr)
+            /* scala/bug#10551: derived value class refs wrapped in existentials
+             * need to be erased the same as non-existential value class refs. */
+            val erasureTarget = ct.typeValue.dealiasWiden match {
+              case ExistentialType(_, underlying) => underlying
+              case tpe => tpe
+            }
```

Comment:
This is wrong: something like
```scala
class AV[X, Y](val xy: (X, Y)) extends AnyVal
type A[X] = AV[X, _]
type Z = A[_]
```
produces types that re-produce the bug. Running partests while I sleep; code to follow coffee tomorrow.